Codeinwp/neve
# fix: change check on decoded json value as per specs

### Summary
When working on Codeinwp/neve#3454 I discovered an issue when the control already has a non-responsive value.
From looking at the `dynamic_selector.php` it looks like it should return the set value if `json_decode()` fails.
I've looked at the specs here: https://www.php.net/manual/en/function.json-decode.php

The return value on failure is `null`.

I changed the condition to reflect that.
With this change, it returns the previously set value instead of `false` for the `string` type.

### Will affect the visual aspect of the product

NO

### Screenshots 

### Test instructions
1. Set a theme mod value for a responsive control to a string (Eg. `wp theme mod set neve_author_box_content_alignment right`)
2. It should return the set value on the frontend until saved again with the new responsive control
3. Check how it behaved previously it would return the default instead, although the value was set.


Closes Codeinwp/neve#3454.
